### PR TITLE
MILESTOME 5: VIEWS

### PR DESCRIPTION
**Pull request by AbdimulhinYussuf3675**
In this branch, I was able to implement the following features :

- [x] Implemented the design from the sneak peek wireframes.
- [x] Used methods that have been created in the [Blog App - processing data in models](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project.md).
- [x] Added some data to the database.

### CODE REVIEW

File: app/views/posts/index.html.erb [hunk 1]
```diff
@@ -1,7 +1,31 @@
...
 
-<% @posts.each do |post|%>
-    <div class="post">
-        <%= post.text %>
+    <div class="user-box">
+        <img class="user-img" src=<%= @user.photo%> alt="user"/>
+        <div class="user-info-box">
+            <h2 class="user-name"><%= @user.name %></h2>
+            <p class="user-post-count">Number of posts: <%= @user.posts_counter%></p>
+
```
Comment (elisha2kyakpo1) [inline app/views/posts/index.html.erb]: - [ ] Kindly note that you are calling a @user (instance variable) in a post view, this is giving me an error, saying undefined method photo for nil class. When you take out this piece of code, the app works as expected. Let's make changes an ask for another review.

post-index

File: app/views/posts/index.html.erb [hunk 2]
```diff
@@ -1,7 +1,31 @@
...
+        <div class="user-recent-post">
+            <h2 class="user-post-title">Post: #<%= post.id %></h2>
+            <p class="user-post-text"><%= post.text %></p>
+            <h3 class="counter">Comments: <%= post.comments_counter%>, Likes: <%= post.likes_counter %><h3>
+
+            <div class="comment_container">
+                <% post.recent_comments.each do |comment|%>
+                    <h3 class="comments"><%= comment.author.name %>:     <%= comment.text %></h3>
+                <% end %>
+            </div>
```
Comment (elisha2kyakpo1) [inline app/views/posts/index.html.erb:13]: - [ ] Kindly follow the wireframe designs, this is clearly outlined in the project requirements. We don't have to do a lot of CSS, but making the pages look like the wireframes is one of the requirements.

design

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 @AbdimulhinYussuf3675, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

👍 Good use of model methods
✔️ The PR title and description are good and professional.
✔️ No linter errors.
✔️ Descriptive readme file.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/posts/index.html.erb (hunk 1 above)
Comment (AbdimulhinYussuf3675) [reply app/views/posts/index.html.erb]: Hello @elisha2kyakpo1, Thanks for the review. Please I don't understand which code am supposed to remove. Can please explain to me further

**Review by mahbubul14 — APPROVED** [review]
# Project Approved :trophy: :tada: 

Hi @AbdimulhinYussuf3675, 

Your project is complete!  You have done it exceptionally well. it's time to merge it :100: 
Congratulations! 🎉 

### To Highlight :+1: 

:heavy_check_mark: No linter error.
:heavy_check_mark: Descriptive readme file.
:heavy_check_mark: Well-implemented suggested changes.

### Optional suggestions :arrow_down_small: 

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better.

* N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also connect with me on [slack](https://microverse-students.slack.com/team/U01J39GL614)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**